### PR TITLE
Add Claude-3 support (text and vision)

### PR DESCRIPTION
**Pull request by juberti**

### CODE REVIEW

File: great_wave.png
Comment (mdepinet) [inline great_wave.png:1]: Does this repo have git lfs set up?

File: llm_benchmark.py [hunk 1]
```diff
@@ -261,28 +261,60 @@ async def openai_embed(context: ApiContext) -> ApiResult:
...
-    }
+    data = make_openai_chat_body(
+        messages=make_anthropic_messages(context.prompt, context.files)
+    )
     return await post(context, url, headers, data, chunk_gen)
 
 
 async def cloudflare_chat(context: ApiContext) -> ApiResult:
+    """Make a Cloudflare chat completion request. The protocol is similar to OpenAI's,
+    but the URL doesn't follow the same scheme."""
```
Comment (mdepinet) [inline llm_benchmark.py:316]:
```suggestion
    but the URL doesn't follow the same scheme and the response structure is different."""

```

File: llm_benchmark.py [hunk 2]
```diff
@@ -395,13 +427,18 @@ async def chunk_gen(response) -> TokenGenerator:
...
+    but the authorization header is X-Api-Key instead of Authorization."""
     url = "https://api.neets.ai/v1/chat/completions"
     headers = make_headers(x_api_key=get_api_key("NEETS_API_KEY"))
     data = make_openai_chat_body(messages=make_openai_messages(context.prompt))
     return await post(context, url, headers, data, openai_chunk_gen)
 
 
 async def together_chat(context: ApiContext) -> ApiResult:
+    """Make a Together chat completion request. The protocol is similar to OpenAI's,
+    but the URL doesn't follow the same scheme."""
```
Comment (mdepinet) [inline llm_benchmark.py:440]:
```suggestion
    but the URL doesn't follow the same scheme and the response structure is slightly different."""

```

File: llm_benchmark.py [hunk 3]
```diff
@@ -261,28 +261,60 @@ async def openai_embed(context: ApiContext) -> ApiResult:
...
-
-    url = "https://api.anthropic.com/v1/complete"
+            delta = chunk.get("delta")
+            if delta and delta.get("type") == "text_delta":
+                tokens += 1
+                yield delta["text"]
+            usage = chunk.get("usage")
+            if usage:
+                num_tokens = usage.get("output_tokens")
+                while tokens < num_tokens:
```
Comment (mdepinet) [inline llm_benchmark.py:297]: Is this worth logging? Seems like it artificially increases TPS

**Review by mdepinet — APPROVED** [review]

File: llm_benchmark.py (hunk 3 above)
Comment (juberti) [reply llm_benchmark.py:297]: It turns out that not every token results in a chunk, so we emit some fake chunks at the end to get to the right number. Ultimately we should just emit this number directly, but the work needed for this seemed like it was going to be more than I wanted to take on in this PR.

File: great_wave.png
Comment (juberti) [reply great_wave.png:1]: hmm, probably not. let me look at that

Comment (juberti) [reply great_wave.png:1]: done

**Pull request merged**